shared/src/publication/Metadata.ts

```diff
@@ -0,0 +1,329 @@
+   *
+   * If the metadata can't be parsed, a warning will be logged with [warnings].
+   */
+  public static deserialize(json: any): Metadata | undefined {
+    if (!(json && json.title)) return;
+
+    const title = LocalizedString.deserialize(json.title) as LocalizedString;
+    const typeUri = json['@type'];
+    const identifier = json.identifier;
+    const subtitle = LocalizedString.deserialize(json.subtitle);
+    const sortAs = LocalizedString.deserialize(json.sortAs);
+    const artists = Contributors.deserialize(json.artist);
+    const authors = Contributors.deserialize(json.author);
+    const colorists = Contributors.deserialize(json.colorist);
+    const contributors = Contributors.deserialize(json.contributor);
+    const editors = Contributors.deserialize(json.editor);
+    const illustrators = Contributors.deserialize(json.illustrator);
+    const inkers = Contributors.deserialize(json.inker);
+    const letterers = Contributors.deserialize(json.letterer);
+    const narrators = Contributors.deserialize(json.narrator);
+    const pencilers = Contributors.deserialize(json.penciler);
+    const translators = Contributors.deserialize(json.translator);
+    const languages = arrayfromJSONorString(json.language);
+    const description = json.description;
+    const publishers = Contributors.deserialize(json.publisher);
+    const imprints = Contributors.deserialize(json.imprint);
+    const published = datefromJSON(json.published);
+    const modified = datefromJSON(json.modified);
+    const subjects = Subjects.deserialize(json.subject);
+    const belongsTo = BelongsTo.deserialize(json.belongsTo);
+    const readingProgression = json.readingProgression;
+    const duration = positiveNumberfromJSON(json.duration);
+    const numberOfPages = positiveNumberfromJSON(json.numberOfPages);
+
+    let otherMetadata = Object.assign({}, json);
+    Metadata.mappedProperties.forEach(x => delete otherMetadata[x]);
+    if (Object.keys(otherMetadata).length === 0) {
+      otherMetadata = undefined;
+    }
+
+    return new Metadata({
+      title,
+      typeUri,
+      identifier,
+      subtitle,
+      sortAs,
+      artists,
+      authors,
+      colorists,
+      contributors,
+      editors,
+      illustrators,
+      inkers,
+      letterers,
+      narrators,
+      pencilers,
+      translators,
+      languages,
+      description,
+      publishers,
+      imprints,
+      published,
+      modified,
+      subjects,
+      belongsTo,
+      readingProgression,
+      duration,
+      numberOfPages,
+      otherMetadata,
+    });
+  }
+
+  /**
+   * Serializes a [Metadata] to its RWPM JSON representation.
+   */
+  public serialize(): any {
+    const json: any = { title: this.title.serialize() };
+    if (this.typeUri !== undefined) json['@type'] = this.typeUri;
+    if (this.identifier !== undefined) json.identifier = this.identifier;
+    if (this.subtitle) json.subtitle = this.subtitle.serialize();
```

Comment:
Another pattern that allows you to remove more of your `if` guards is the use of the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator:

```suggestion
    json.subtitle = this.subtitle?.serialize();
```